plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/cache/JDBCBasicDataTypeCache.java

```diff
@@ -67,8 +68,10 @@ protected OBJECT fetchObject(@NotNull JDBCSession session, @NotNull OWNER owner,
             return null;
         }
         int valueType = JDBCUtils.safeGetInt(dbResult, JDBCConstants.DATA_TYPE);
-        // Check for bad value type for strings: #494
-        if (valueType == Types.BINARY && (name.contains("varchar") || name.contains("VARCHAR"))) {
+        // Check for bad value type for strings: #494 && 19219
+        if ((valueType == Types.BINARY  && name.toUpperCase(Locale.ENGLISH).contains("VARCHAR")) ||
+            valueType == Types.JAVA_OBJECT && SQLConstants.DATA_TYPE_VARCHAR.equalsIgnoreCase(name)

```

Comment:
Then let's be good boys and clarify our intentions:

```suggestion
            (valueType == Types.JAVA_OBJECT && SQLConstants.DATA_TYPE_VARCHAR.equalsIgnoreCase(name))

```